### PR TITLE
Added delete button

### PR DESCRIPTION
**Pull request by hugolundin**
Closing issue #54. 
- Added an is_active-property for users, which is changed to False when a user is deleted from the admin ui.
- Added a button for removing users from the admin ui, which uses ajax and a route in views.py.
- Added a new database migration.
- Added a function which prevents the user from closing/reloading the website when data is not saved.

### CODE REVIEW

**Comment by Limpan** [comment]
Rebase, since #91 breaks it.

File: web/app/templates/base.html [hunk 1]
```diff
@@ -29,4 +29,7 @@
   {{ super() }}
   {{ moment.include_moment() }}
   <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script>
+  <script type="text/javascript">
+    var $SCRIPT_ROOT = {{ url_for('main.index', _external = True)|tojson|safe }}
```
Comment (Greenheart) [inline web/app/templates/base.html:33]: This could be solved with some property from `window.location`

Comment (hugolundin) [inline web/app/templates/base.html:33]: Nej

File: web/app/models.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ class User(UserMixin, db.Model):
     first_name = db.Column(db.String(64))
     last_name = db.Column(db.String(64))
     password_hash = db.Column(db.String(128), default=None)
+    is_active = db.Column(db.Boolean, unique=False, default=True)
```
Comment (Greenheart) [inline web/app/models.py]: Unique is set to false by default

File: web/app/static/js/prevent-leave.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+$(document).ready(function () {
+    'use strict';
+    var first_name = $('#first_name').val();
```
Comment (Greenheart) [inline web/app/static/js/prevent-leave.js]: Otiose semicolon ;)

File: web/app/static/js/prevent-leave.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    window.onbeforeunload = function () {
+
+        var preventLeave = false;
+
+        if ($('#first_name').val() !== first_name) {
+            preventLeave = true;
+        }
+
+        if ($('#last_name').val() !== last_name) {
+            preventLeave = true;
```
Comment (Greenheart) [inline web/app/static/js/prevent-leave.js]: ;

File: web/app/static/js/delete-teacher.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        'animate': true,
+        'buttons': {
+            'yes': {
+                'label': 'Ja',
+                'className': 'btn-success fixed-width-button',
+                'callback': function () {
+                  var url = $SCRIPT_ROOT + 'admin/teachers/remove'
+                  var data = { 'user': userId }
+
+                  $.ajax({type: 'POST', url: url, data: JSON.stringify(data), contentType: 'application/json',
```
Comment (Greenheart) [inline web/app/static/js/delete-teacher.js:22]: `$.post()` is a shorthand for $.ajax({type: 'POST'}) that does exactly the same thing.

Might be a future improvement

**Pull request merged**